### PR TITLE
Bump the kafka version to 3.2.0 and vertx version to 4.3.1

### PR DESCRIPTION
**Pull request by ShubhamRwt**
Signed-off-by: ShubhamRwt <srawat@redhat.com>

This PR bumps the Kafka version in the bridge to 3.2.0 since vertx 4.3.1 is released and is compatible to work with kafka version >= 3.0.0 also.The PR replaces the deprecated `routingContext.getbody()` methods with the new alternatives and also other fixes

### CODE REVIEW

File: src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java [hunk 1]
```diff
@@ -276,7 +276,7 @@ private void createConsumer(RoutingContext routingContext) {
         this.httpBridgeContext.setOpenApiOperation(HttpOpenApiOperations.CREATE_CONSUMER);
 
         // check for an empty body
-        JsonObject body = routingContext.getBody() != null && routingContext.getBody().length() != 0 ? routingContext.getBodyAsJson() : new JsonObject();
+        JsonObject body = routingContext.body().buffer() != null && routingContext.body().buffer().length() != 0 ? routingContext.getBodyAsJson() : new JsonObject();
```
Comment (tomncooper) [inline src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java:279]: So has the new vertx version changed the inplementation of RoutingContext?

File: src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java [hunk 2]
```diff
@@ -502,8 +502,8 @@ public void handle(Endpoint<?> endpoint, Handler<?> handler) {
         JsonObject bodyAsJson = EMPTY_JSON;
         try {
             // check for an empty body
-            if (routingContext.getBody() != null && routingContext.getBody().length() != 0) {
-                bodyAsJson = routingContext.getBodyAsJson();
+            if (routingContext.body().buffer() != null && routingContext.body().buffer().length() != 0) {
+                bodyAsJson = routingContext.body().asJsonObject();
```
Comment (tomncooper) [inline src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java:506]: Everywhere else you are using `.body().buffer().....` why do you not need that here? Dos the json method handle that for you?

File: src/test/java/io/strimzi/kafka/bridge/http/ConsumerIT.java [hunk 3]
```diff
@@ -475,7 +475,7 @@ void createConsumerWithWrongFetchMinBytes(VertxTestContext context) throws Inter
     @Test
     void createConsumerWithNotExistingParameter(VertxTestContext context) throws InterruptedException, TimeoutException, ExecutionException {
         checkCreatingConsumer("foo", "bar", HttpResponseStatus.BAD_REQUEST,
-                "Validation error on:  - provided object should not contain additional properties", context);
+                "Validation error on:  - Provided object contains unexpected additional property: foo", context);
```
Comment (tomncooper) [inline src/test/java/io/strimzi/kafka/bridge/http/ConsumerIT.java:478]: Not sure "foo" should still be in there?

File: src/test/java/io/strimzi/kafka/bridge/http/HttpCorsIT.java [hunk 4]
```diff
@@ -110,6 +110,7 @@ public void testCorsNotEnabled(VertxTestContext context) {
                         client.request(HttpMethod.POST, 8080, "localhost", "/consumers/1/instances/1/subscription")
                                 .putHeader("Origin", "https://evil.io")
                                 .send(ar2 -> context.verify(() -> {
+                                    System.out.println(ar2.result().body());
```
Comment (tomncooper) [inline src/test/java/io/strimzi/kafka/bridge/http/HttpCorsIT.java:113]: You need to make sure you are removing these debug print lines

**Review by tomncooper — CHANGES_REQUESTED** [review]
Just a few nits to fix

File: src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java (hunk 2 above)
Comment (ShubhamRwt) [reply src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java:506]: The alternative suggested for routingContext.getBodyasJson() was routingContext.body().asJsonObject().

File: src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java (hunk 1 above)
Comment (ppatierno) [reply src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java:279]: Yeah they have deprecated the `getBody` and other methods, everything seems to be under a `RequestBody` object returned by `body` method.

File: src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java (hunk 2 above)
Comment (ppatierno) [reply src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java:506]: As I can see from the current implementation here https://github.com/vert-x3/vertx-web/blob/master/vertx-web/src/main/java/io/vertx/ext/web/RequestBody.java#L131 you could change the if condition just checking `rounting.body().isEmpty()` instead of checking null and length on it.

File: src/test/java/io/strimzi/kafka/bridge/http/HttpCorsIT.java [hunk 5]
```diff
@@ -180,8 +179,8 @@ public void testCorsOriginAllowed(VertxTestContext context) {
                                 .putHeader("Origin", "https://strimzi.io")
                                 .putHeader("content-type", BridgeContentType.KAFKA_JSON)
                                 .sendJsonObject(topicsRoot, ar2 -> context.verify(() -> {
-                                    //we are not creating a topic, so we will get a 404 status code
-                                    assertThat(ar2.result().statusCode(), is(404));
+                                    //we are not creating a topic, so we will get a 400 status code
+                                    assertThat(ar2.result().statusCode(), is(400));
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/HttpCorsIT.java:183]: Why this change?

File: src/main/java/io/strimzi/kafka/bridge/http/HttpAdminClientEndpoint.java [hunk 6]
```diff
@@ -156,7 +156,7 @@ public void doGetTopic(RoutingContext routingContext) {
                 root.put("partitions", partitionsArray);
                 HttpUtils.sendResponse(routingContext, HttpResponseStatus.OK.code(), BridgeContentType.KAFKA_JSON, root.toBuffer());
 
-            } else if (done.cause() instanceof UnknownTopicOrPartitionException) {
+            } else if (done.cause() != null && done.cause().getCause() instanceof UnknownTopicOrPartitionException) {
```
[the remark this answers is not in the document]
Comment (ShubhamRwt) [reply src/main/java/io/strimzi/kafka/bridge/http/HttpAdminClientEndpoint.java:159]: done.cause() was returning `CompletionException` instead of the `UnknownTopicOrPartitionException` or you can also say that with the latest vertx and kafka, the `UnknownTopicOrPartitionException` comes wrapped inside the `CompletionException`. So I had to use done.cause().getCause() get the `UnknownTopicOrPartitionException` out of the done.cause() which was `CompletionException`

**Review by ppatierno — APPROVED** [review]
